src/java.base/share/classes/java/io/FileDescriptor.java

```diff
@@ -59,15 +61,35 @@ public final class FileDescriptor {
     private List<Closeable> otherParents;
     private boolean closed;
 
+    /**
+     * Called by FileDispatcherImpl when the file descriptor is about to be closed natively.
+     */
+    public void markClosedByNIO() {
```

Comment:
Can we avoid changing the public interface? This is not a method we'd like users to call

Reply:
I've checked if it would be possible to extend FileDescriptor and put this method there, but there's several non-trivial places where this would have to be changed (e.g. in `sun.nio.ch.InheritedChannel` the FD is constructed using a non-public ctor via reflection). So we can make this package-private and use reflection, or probablz a better way would be through `JavaIOFileDescriptorAccess`.